src/lib.rs

```diff
@@ -1,2 +1,65 @@
+use anyhow::{Result, bail};
+use log::{info};
+
 mod arti;
-mod ffi;
\ No newline at end of file
+mod ffi;
+
+/// Request for the different REST methods. Here we don't differentiate between
+/// the url and the parameters passed to the remote end.
+/// Only http and https URLs are supported for the moment.
+pub struct Request {
+    pub url: String,
+    pub user_agent: Option<String>,
```

Comment:
So that would be a list of key/values then? I'm a bit afraid of doing complicated structures to pass back and forth rust and mobile. It's my first mobile rust library ;)

My current approach is: make something as simple as possible, but not simpler... So if you think that should include a list of generic headers, add it. If the current SwissCovid / NotifyMe apps don't use anything else than the `user_agent`, I'll keep it like that. Makes sense?

Reply:
> So that would be a list of key/values then? I'm a bit afraid of doing complicated structures to pass back and forth rust and mobile. It's my first mobile rust library ;)

We should be able to assess the supported types, and make some tests. For the Java side (JNI), lists and maps are supported: https://docs.rs/jni/0.19.0/jni/all.html .

> My current approach is: make something as simple as possible, but not simpler... So if you think that should include a list of generic headers, add it. If the current SwissCovid / NotifyMe apps don't use anything else than the `user_agent`, I'll keep it like that. Makes sense?

Then in that spirit, do we then need PUT and DELETE? Does NotifyMe use them?

Reply:
Correct. I think only `POST` is needed in fact, not even `GET`.